apps/nextjs/src/app/api/chat/chatHandler.ts

```diff
@@ -287,9 +289,13 @@ export async function handleChatPostRequest(
               ],
               threatDetectors: () => threatDetectors,
             },
-
             document: {
               content: dbLessonPlan ?? {},
+              schema: LessonPlanSchema,
```

Comment:
This is the main starting point. We want to pass in a schema that we want to work with.